webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -610,3 +628,32 @@ def _get_first_observation_x(obs_data: pd.DataFrame) -> Union[int, str]:
         )
 
     return caster.get(type(first_observation), lambda *args: False)(first_observation)
+
+
+def _format_index_value(axis: pd.Index, index: int) -> Union[np.number, datetime.date]:
+    """_format_index_value takes the value of `axis` at position `index` and
+    tries to convert it to a datetime. If this works, it returns just the date.
+    If parsing fails, it returns the value unaltered."""
+    raw_value = axis[index]
+    if isinstance(raw_value, str):
+        try:
+            datetime_value = pd.to_datetime(raw_value)
+            return datetime_value.date()
+        except (pd.errors.ParserError, ValueError):
+            pass
+    return raw_value
+
+
+def _style_scatterplot(
+    fig: go.Figure,
+    active_response: str,
+    active_parameter: str,
+    x_value: Union[datetime.date, np.number],
+    n_rows: int,
+) -> None:
+    fig.update_layout(assets.ERTSTYLE["figure"]["layout"])
+    fig.update_xaxes(title_text=f"{active_response} @ {x_value}", row=1, col=1)
+    fig.update_yaxes(title_text=f"{active_parameter}", row=1, col=1)
+    fig.update_xaxes(title_text=f"{active_parameter}", row=n_rows, col=1)
+    fig.update_xaxes(title_text=f"{active_response}", row=n_rows, col=2)
+    fig.update_layout(showlegend=False)
```

Comment:
In other parts of the code, we have updated the layout via a dictionary and not passing the `fig` element, e.g. `_define_style_ensemble` or `_layout_figure`. This was chosen to have simpler variables been passed and consequently easier to test if we want so. What would you think about maintaining a similar pattern for layout functions?

Reply:
are you referring just to the last line, 659, setting visibility of the legend? or also to the `update_?axes` calls?

if you mean the latter, I can try to see if this can be replicated by passing a dict to `update_layout`

Reply:
i tried to take a look. following assumptions were made: using the `update_layout` function, we need to pass a dict (cf. [docs on `update_layout`)[https://plotly.com/python-api-reference/generated/plotly.graph_objects.Figure.html?highlight=update_layout#plotly.graph_objects.Figure.update_layout]. that dict should be of type [`plotly.graph_objects.layout`](https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Layout.html#plotly.graph_objects.Layout)

Now, that one has a property `Grid` where one can supply `xaxes`, but that one should be a list of strings matching some regex (`^x([2-9]|[1-9][0-9]+)?( domain)?$`), which among other things means "starts with x" - this does not seem to be what we need. also, the docs for that property mention

> Used with yaxes when the x and y axes are shared across columns and rows

so again this doesn't seem to be what we need, as we don't want to have axes shared across columns and rows

trying to set `xaxes` directly  in the layout dict (without `grid`) gives an error message.

it looks like the `update_?axes` methods cannot be replaced with a layout dict : /

Reply:
I understand. Thank you for verifying that it is not possible and writing it out!

Reply:
no problem! : )